File: mrz-rn-library/ios/MrzCameraViewManager.m
//
//  MrzCameraViewManager.m
//  MRZLibrary
//
//  Objective-C bridge that registers the Swift MrzCameraViewManager with
//  React Native and exposes its props / events.
//

#import <React/RCTViewManager.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTUIManager.h>

@interface RCT_EXTERN_MODULE(MrzCameraViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(onMrzDetected, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onMrzError, RCTBubblingEventBlock)

RCT_EXPORT_VIEW_PROPERTY(focusViewBorderColor, NSString)
RCT_EXPORT_VIEW_PROPERTY(focusViewStrokeWidth, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(paused, BOOL)

@end

File: mrz-rn-library/ios/MrzCameraViewManager.swift
//
//  MrzCameraViewManager.swift
//  MRZLibrary
//
//  RCTViewManager that exposes MrzCameraView to JS as <MrzCameraView/>.
//

import Foundation
import UIKit

@objc(MrzCameraViewManager)
class MrzCameraViewManager: RCTViewManager {

    override static func requiresMainQueueSetup() -> Bool {
        return true
    }

    override func view() -> UIView! {
        return MrzCameraView()
    }
}
